pkg/rpc/heartbeat.go

```diff
@@ -56,6 +58,16 @@ type HeartbeatService struct {
 	testingAllowNamedRPCToAnonymousServer bool
 }
 
+func checkClusterName(clusterName string, peerName string) error {
+	if clusterName != peerName {
+		err := errors.Errorf(
+			"local cluster name %q does not match peer cluster name %q", clusterName, peerName)
```

Comment:
I'm wondering if we want clearer error messages in the case of `peer cluster name is %q, but local cluster name is not set` (and vice versa). The quoted name should be sufficient, so up to you.

Reply:
Done.